Expanding node info cards

jamesarich opened the pull request:
This is a rework of the UI for NodeInfo cards to be expandable. Tapping the card now expands the view to show less pertinent details as discussed in discord, https://github.com/meshtastic/Meshtastic-Android/pull/1111#issuecomment-2197270357, and #1138. A global show details toggle is available in the filter menu.
As part of this -  the views were reworked from a Constraint view into Rows for future readability and maintainability.

Hides the coordinates by default as mentioned in @garthvh's comment here https://github.com/meshtastic/Meshtastic-Android/discussions/1138#discussioncomment-10046245

Adds the node id when available - resolves  #1097.

Before
<img width="407" alt="image" src="https://github.com/user-attachments/assets/d4d79ea0-af97-4c0c-8e7e-07c8511eddc7">


After
<img width="399" alt="image" src="https://github.com/user-attachments/assets/a9a2600f-a671-4399-b9c1-5e2e9ee72b27">


Also adds a global Show Details toggle to the Filter Menu. 
<img width="349" alt="image" src="https://github.com/user-attachments/assets/84b6f5d6-2df1-4ea7-8d9b-c940aee9b31f">

b8b8 commented:
Can this be locked in expanded view?

jamesarich commented:
> Can this be locked in expanded view?

it's currently just toggle-able per card

b8b8 commented:
ok, can you make it lock open if we want. some of us want to see all the data all the time by just scrolling vertically instead of having to tap each node. But so far this looks great. Moving the coordinate down to expanded view but keeping all other information visible in collapsed view was the right call. now we aren't relegating important info the only the expanded view while still having lots of room for future data to not impede the look.

b8b8 commented:
no worries, I just would like it to be a future consideration/option.
There's been a lot of talk of hiding all the info behind separate cards, and stripping away the node info page to only the most basic info. I think the way shown here is the best solution. Not everything needs to be dumbed down.

jamesarich commented:
@b8b8 - added a global toggle in the filter menu

b8b8 commented:
THANK YOU

jamesarich commented:
fixed expand toggle logic, added longpress to copy gps coordinates
<img width="349" alt="image" src="https://github.com/user-attachments/assets/e0131565-2788-4591-84e3-312b538985de">

jamesarich commented:
added node info text selection
<img width="349" alt="image" src="https://github.com/user-attachments/assets/6abd2a17-57f5-4d30-9373-291d407c961b">

andrekir commented on app/src/main/java/com/geeksville/mesh/ui/NodeInfo.kt:
```diff
@@ -74,222 +95,172 @@ fun NodeInfo(
+                                .defaultMinSize(minHeight = 32.dp)
+                                .clickable {
+                                    chipClicked()
+                                },
```
.clickable can be removed here

andrekir commented:
we no longer need `contraintlayout-compose` dependency, please remove:

https://github.com/meshtastic/Meshtastic-Android/blob/de180f2d3556a2e8338099450c2f9a4d8c9f5972/app/build.gradle#L196

andrekir commented on app/src/main/java/com/geeksville/mesh/ui/SignalInfo.kt:
```diff
@@ -17,11 +17,14 @@ fun signalInfo(
+                metrics.channelUtilization,
+                metrics.airUtilTx
+            )
+        } ?: ""
```
I don't think this is necessary. device metrics should never be null. if it is, this makes it clear to users something is wrong.

andrekir reviewed (approved):
looks good, thanks!
